### ISSUE TITLE
[doc] Provide source links to documentation

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#3269]
All Keras methods have a "source" link (as does/did? Numpy). 

It would be good to add this to Ray documentation.

Example: https://keras.io/layers/merge/

**Comment by tashachin** [comment]
Hi, I'd like to help with this issue.

Where would I be fetching and inserting hyperlinks? I'm not 100% sure I understand the endpoint architecture or the needs of your project.

**Comment by richardliaw** [comment]
Thanks for offering @tashachin! I just took a quick look into actually doing this, and it might not be too hard to do. 

Ideally, we'd have for all of these methods on this page (https://ray.readthedocs.io/en/latest/tune-package-ref.html), a link to `[source]` that looks like 
![test](https://user-images.githubusercontent.com/4529381/48170989-c3163700-e2ae-11e8-8edf-425697f42f9c.png)

From this thread, https://github.com/rtfd/readthedocs.org/issues/2139#issuecomment-390287246, it seems like you can just modify a configuration script parameter so that this is the case?

**Comment by tashachin** [comment]
Hey @richardliaw I have a potential solve but I'm not sure how to run tests on my local machine.

I noticed that in your [conf.py](https://github.com/tashachin/ray/blob/master/doc/source/conf.py), line 220 is commented out:

```
# If true, links to the reST sources are added to the pages.
#html_show_sourcelink = True
```

Is this intentional by design? I forked the repo and am ready to submit a PR to switch this flag, but I'm concerned it'd affect your project on a much wider scale than is necessary.

**Comment by richardliaw** [comment]
Hey; you can just run `make html` and if it doesn't throw any warnings, you'll probably have passed the test.

I think that line is not intentional; switching it might be fine.

Feel free to post the PR! I'll work with you to fix things and get it merged.

**Comment by tashachin** [comment]
Hey @richardliaw; I won't have access to my Macbook for a month or so. I'd like to let someone else handle this issue in my stead.

Thanks for your patience and for letting me attempt this one.

**Comment by richardliaw** [comment]
This is addressed in #3281.
